src/mbgl/map/transform_state.hpp

```diff
@@ -74,13 +77,17 @@ class TransformState {
     double zoomScale(double zoom) const;
     double scaleZoom(double scale) const;
 
+    constexpr explicit operator bool() const {
```

Comment:
> Let's not use operator bool() for validity checks. 

We've discussed this before, yet we still use `explicit operator bool` in other parts of the code e.g. [Size::operator bool](https://github.com/mapbox/mapbox-gl-native/blame/94f011895c8e1bde36ee2ec235dbbcf2c994ac4c/include/mbgl/util/size.hpp#L20).

I don't have a strong opining about replacing it with something like `valid()`, however I'd prefer to have a separate PR that replaces every `explicit operator bool` usage for validation in our code 👍